### PR TITLE
fix: add check for negative fluorescence values (#101)

### PR DESCRIPTION
**Pull request by maxschloegel**
Due to baseline deviations from 0, negative maximum
fluorenscence values might be introduced. This leads
to issues when plotting the data, since often a
log scale is used. To notify about negative fl-values
new dataset-checks were implemented.

### CODE REVIEW

File: dclab/rtdc_dataset/check.py [hunk 1]
```diff
@@ -391,6 +391,36 @@ def check_fl_samples_per_event(self, **kwargs):
                             cfg_key="samples per event"))
         return cues
 
+    def check_fl_max_positive(self, **kwargs):
+        """ Check if all fl?_max values are >0.1 """
```
Comment (paulmueller) [inline dclab/rtdc_dataset/check.py:395]: please remove the spaces at the beginning and end of the docstring

File: dclab/rtdc_dataset/check.py [hunk 2]
```diff
@@ -391,6 +391,36 @@ def check_fl_samples_per_event(self, **kwargs):
...
+        cues = []
+        neg_feats = []
+        for fl in ['fl1_max', 'fl2_max', 'fl3_max']:
+            if fl in self.ds:
+                if min(self.ds[fl]) <= 0.1:
+                    neg_feats.append(fl)
+        if neg_feats:
+            cues.append(ICue(
+                msg="Negative value for feature {}".format(neg_feats),
```
Comment (paulmueller) [inline dclab/rtdc_dataset/check.py:404]: To make this visually more appealing, I would write
`"Negative value for feature(s): {}".format(", ".join(neg_feats))`

File: tests/test_rtdc_check_dataset.py [hunk 3]
```diff
@@ -291,6 +292,44 @@ def test_ic_metadata_empty_string():
     assert cues[0].cfg_key == "medium"
 
 
+def test_ic_fl_max():
```
Comment (paulmueller) [inline tests/test_rtdc_check_dataset.py:295]: Could you please add one more test where there are no negative values, but values with "0.1"?

**Review by paulmueller — CHANGES_REQUESTED** [review]
Thanks a lot! Just a few minor changes.

**Review by paulmueller — APPROVED** [review]
